pkg/executor/aggregate/agg_util.go

```diff
@@ -213,12 +217,53 @@ func (w *AggWorkerStat) Clone() *AggWorkerStat {
 	}
 }
 
-// ActionSpill returns a AggSpillDiskAction for spilling intermediate data for hashAgg.
-func (e *HashAggExec) ActionSpill() *AggSpillDiskAction {
-	if e.spillAction == nil {
-		e.spillAction = &AggSpillDiskAction{
-			e: e,
+// ActionSpill returns an action for spilling intermediate data for hashAgg.
+func (e *HashAggExec) ActionSpill() memory.ActionOnExceed {
+	if e.IsUnparallelExec {
+		if e.spillAction == nil {
+			e.spillAction = &AggSpillDiskAction{
+				e: e,
+			}
 		}
+		return e.spillAction
+	}
+
+	e.parallelAggSpillAction = &ParallelAggSpillDiskAction{
```

Comment:
Why unparallel exec need to check if `e.spillAction` is nil but parallel exec don't need to do the check?

Reply:
> Why unparallel exec need to check if `e.spillAction` is nil but parallel exec don't need to do the check?

I will delete the check in unparallel exec.

Reply:
> Why unparallel exec need to check if `e.spillAction` is nil but parallel exec don't need to do the check?

I will delete the check in unparallel mode.